packages/contracts-bedrock/deploy/017-L2OutputOracleInit.ts

```diff
@@ -0,0 +1,40 @@
+import assert from 'assert'
+
+import { ethers } from 'ethers'
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import { awaitCondition } from '@eth-optimism/core-utils'
+import '@eth-optimism/hardhat-deploy-config'
+import 'hardhat-deploy'
+
+import { getDeploymentAddress } from '../src/deploy-utils'
+
+const deployFn: DeployFunction = async (hre) => {
+  const proxyAdmin = await getDeploymentAddress(hre, 'ProxyAdmin')
+  const l2OutputOracle = await getDeploymentAddress(hre, "L2OutputOracle")
+  const l2OutputOracleProxy = await getDeploymentAddress(hre, 'L2OutputOracleProxy')
+
+  var startingBlockNumber = hre.deployConfig.l2OutputOracleStartingBlockNumber;
+  var startingTimestamp = hre.deployConfig.l2OutputOracleStartingTimestamp;
+
+  await proxyAdmin.upgradeAndCall(
+    l2OutputOracleProxy,
+    l2OutputOracle,
+    l2OutputOracle.interface.encodeFunctionData('initialize', [startingBlockNumber, startingTimestamp])
+  )
+
+  await awaitCondition(
+    async () => {
+      return (
+        (await l2OutputOracleProxy.callStatic.startingBlockNumber()) === startingBlockNumber
+        &&
+        (await l2OutputOracleProxy.callStatic.startingTimestamp()) === startingTimestamp
+      )
+    },
+    30000,
+    1000
+  )
+}
+
+deployFn.tags = ['ProxyAdmin', 'L2OutputOracle', 'l1']
```

Comment:
perhaps remove `ProxyAdmin` from each of these and change `L2OutputOracle` to `L2OutputOracleInitialize`